test/manager/TaskManagerTest.java

```diff
@@ -0,0 +1,337 @@
+        taskManager.createEpic(epic1);
+        Subtask subtask1InEpic1 = new Subtask("First subtask in epic 1", "First subtask description", Status.NEW, epic1.getId(), LocalDateTime.now(), Duration.ofHours(1));
+        taskManager.createSubtask(subtask1InEpic1);
+        taskManager.removeEpicById(epic1.getId());
+        assertFalse(taskManager.getAllEpics().contains(epic1), "Epic wasn't deleted");
+        assertFalse(taskManager.getAllSubtasks().contains(subtask1InEpic1), "Subtask wasn't deleted");
+        assertFalse(taskManager.getHistory().contains(epic1), "Epic wasn't deleted from history");
+        assertFalse(taskManager.getHistory().contains(subtask1InEpic1), "Subtask wasn't deleted from history");
+
+    }
+
+    @DisplayName("Check subtask removal by id")
+    @Test
+    void shouldRemoveSubtaskById() {
+        Epic epic1 = new Epic("First epic", "First epic description");
+        taskManager.createEpic(epic1);
+        Subtask subtask1InEpic1 = new Subtask("First subtask in epic 1", "First subtask description", Status.NEW, epic1.getId(), LocalDateTime.now(), Duration.ofHours(1));
+        taskManager.createSubtask(subtask1InEpic1);
+        taskManager.removeSubtaskById(subtask1InEpic1.getId());
+        assertFalse(taskManager.getAllSubtasks().contains(subtask1InEpic1), "Subtask wasn't deleted");
+        assertFalse(taskManager.getHistory().contains(subtask1InEpic1), "Subtask wasn't deleted from history");
+    }
+
+    @DisplayName("Remove all epics")
+    @Test
+    void shouldRemoveAllEpics() {
+        Epic epic1 = new Epic(1, "First epic", "First epic description");
+        taskManager.createEpic(epic1);
+        taskManager.getEpicById(epic1.getId());
+        taskManager.removeAllEpics();
+        assertTrue(taskManager.getAllEpics().isEmpty(), "Epics still present");
+        assertTrue(taskManager.getAllSubtasks().isEmpty(), "Subtasks srill present");
+        assertFalse(taskManager.getHistory().contains(epic1), "History is filled");
+    }
+
+    @DisplayName("Update epic")
+    @Test
+    void shouldUpdateEpic() {
+        Epic epic1 = new Epic("First epic", "First epic description");
+        taskManager.createEpic(epic1);
+        epic1.setTitle("New title");
+        epic1.setDescription("New description");
+        taskManager.updateEpic(epic1);
+        Epic updatedEpic = taskManager.getEpicById(epic1.getId());
+        assertNotNull(updatedEpic, "Epic with id " + epic1.getId() + " not found");
+        assertEquals(epic1.getTitle(), updatedEpic.getTitle(), "Title wasn't updated");
+        assertEquals(epic1.getDescription(), updatedEpic.getDescription(), "Description wasn't updated");
+        assertTrue(taskManager.getAllEpics().contains(updatedEpic), "Updated epic not found in the list of all epics");
+    }
+
+    @DisplayName("Update subtasks and epic status")
+    @Test
+    void shouldUpdateSubtasksAndEpicStatus() {
+        Epic epic1 = new Epic("First epic", "First epic description");
```

Comment:
Чтобы каждый раз не создавать одни и теже объекты, нужно выделить их создание в отдельный утильный класс.
Например, для данной строки получится метод
```suggestion
        Epic epic1 = TestUtil.createEpic();
```
А еще нужно стараться максимально избавляться от дублирований, поэтому лучше создать несколько объектов в `BeforeEach` или на уровне переменных класса (например) и переиспользовать их в каждом из тестов